src/core/trulens/core/utils/threading.py

```diff
@@ -125,6 +125,11 @@ class TP(SingletonPerName):  # "thread processing"
     DEBUG_TIMEOUT: Optional[float] = 600.0  # [seconds], None to disable
     """How long to wait (seconds) for any task before restarting it."""
 
+    def __new__(cls) -> TP:
```

Comment:
Why make this change?

Reply:
Whatever is the reason, please add documentation for the next person who will be confused by this.